core/metacling/src/TClingClassInfo.cxx

```diff
@@ -1267,6 +1267,10 @@ const char *TClingClassInfo::Name() const
    if (const NamedDecl* ND = llvm::dyn_cast<NamedDecl>(fDecl)) {
       PrintingPolicy Policy(fDecl->getASTContext().getPrintingPolicy());
       llvm::raw_string_ostream stream(buf);
+
+      // This should be here because getNameForDiagnostic triggers deserialization
```

Comment:
`// getNameForDiagnostic can trigger deserialization` is sufficient; similarly for the other comments. Thanks for putting the comment, though!